aws/logs_monitoring/lambda_function.py

```diff
@@ -287,6 +292,18 @@ def s3_handler(event, context, metadata):
             yield structured_line
 
 
+# Handle CloudWatch logs from Kinesis
+def kinesis_awslogs_handler(event, context, metadata):
+    def reformat_record(record):
+        return {
+            "awslogs": {
+                "data": record["kinesis"]["data"]
+            }
+        }
+        
+    return itertools.chain.from_iterable(awslogs_handler(reformat_record(r), context, metadata) for r in event["Records"])
```

Comment:
I'm little bit new at python but I'm not sure to understand the need of `itertools.chain.from_iterable`, which seems to chain several iterators, as we produce a single iterator with `awslogs_handler(reformat_record(r), context, metadata) for r in event["Records"]`. Am I missing something? WDYT @gaetan-deputier?

Reply:
In python when do you f(x) for x in A, it generates an iterator of values which are the result of each f(x) call (read more here: https://wiki.python.org/moin/Generators). Because the kinesis stream batch will contain multiple awslogs records, we need to iterate over the batch and process each one.

Doing it like this maintains the streaming nature of process. The output of this function is basically a flat iterator over all the awslog records across all of the kinesis records.

Reply:
Thank you @jmc-rival for the explanation, I totally missed that the `awslogs_handler ` is returning an iterator. 👌